### PR TITLE
Generate's operation message_protocols fails for older semantics libs

### PR DESCRIPTION
**Pull request by vishnu-alapati**
### Description of the Change
Made changes to /message_protocols end-point to work for old REMRem-Semantics library. As the method getProtocolEdition() is created/implemented on semantics 2.2.0, if we use semantics library prior to that the /message_protocols end point is not working properly we are seeing 500 status error.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Vishnu Alapati vishnu.alapati@tcs.com

### CODE REVIEW

File: service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java [hunk 1]
```diff
@@ -267,7 +267,12 @@ public JsonElement getMessageProtocols() {
         for (MsgService service : msgServices) {
             JsonObject protocolObject = new JsonObject();
             protocolObject.addProperty("name", service.getServiceName());
-            protocolObject.addProperty("edition", service.getProtocolEdition());
+            try {
+                protocolObject.addProperty("edition", service.getProtocolEdition());
+            } catch (NoSuchMethodError | AbstractMethodError e) {
+                // Ignored intentionally in order to ensure compatibility with
```
Comment (z-sztrom) [inline service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java:273]: Please, add a warning to log file to clearly indicate that the action failed due to an old used library and suggest upgrade to a fresh one.

Comment (vishnu-alapati) [reply service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java:273]: Sure Roman

File: service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java [hunk 2]
```diff
@@ -267,7 +267,13 @@ public JsonElement getMessageProtocols() {
...
+                // eiffel-remrem-semantics:2.2.0 and older.
+                log.warn("Failed due to older version of REMRem Semantics library, please use the library which is greater than 2.2.0 version.");
```
Comment (z-sztrom) [inline service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java:275]: I think the message is confusing as it mentions Semantics library. But how do you know that it's a semantics library? The problematics method is getProtocolEdition() and it's defined in eiffel-remrem-protocol-interface:2.1.2. So the message should be something like 'An old library, without implementation of MsgService.getProtocolEdition(), is used. Please, upgrade to a newer library implementing eiffel-remrem-protocol-interface:2.1.2 or higher.' The message should be logged as 'error' to indicate that the upgrade is really needed.

**Review by z-sztrom — CHANGES_REQUESTED** [review]

File: service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java (hunk 2 above)
Comment (vishnu-alapati) [reply service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java]: Updated the error message

**Review by z-sztrom — APPROVED** [review]
Works fine. Older versions of libraries are handled correctly.

**Review by piyush-sadangi — APPROVED** [review]
looks good to me.

**Review by jainadc9 — APPROVED** [review]

**Pull request merged**